cpp/src/io/json/json_normalization.cu

```diff
@@ -354,5 +466,114 @@ void normalize_whitespace(datasource::owning_buffer<rmm::device_buffer>& indata,
   std::swap(indata, outdata);
 }
 
+std::
+  tuple<rmm::device_uvector<char>, rmm::device_uvector<size_type>, rmm::device_uvector<size_type>>
+  mixed_type_column_ws_normalization(device_span<char const> d_input,
+                                     rmm::device_uvector<size_type>& col_lengths,
+                                     rmm::device_uvector<size_type>& col_offsets,
+                                     rmm::cuda_stream_view stream,
+                                     rmm::device_async_resource_ref mr)
+{
+  size_t col_lengths_size = col_lengths.size();
+  size_type inbuf_size =
+    thrust::reduce(rmm::exec_policy_nosync(stream), col_lengths.begin(), col_lengths.end());
+  rmm::device_uvector<char> inbuf(inbuf_size, stream);
+  rmm::device_uvector<size_type> inbuf_offsets(col_lengths_size, stream);
+  thrust::exclusive_scan(rmm::exec_policy_nosync(stream),
+                         col_lengths.begin(),
+                         col_lengths.end(),
+                         inbuf_offsets.begin(),
+                         0);
+
+  auto input_it = thrust::make_transform_iterator(
+    thrust::make_counting_iterator(0),
+    cuda::proclaim_return_type<char const*>(
+      [d_input = d_input.begin(), col_offsets = col_offsets.begin()] __device__(
+        size_t i) -> char const* { return &d_input[col_offsets[i]]; }));
+  auto output_it = thrust::make_transform_iterator(
+    thrust::make_counting_iterator(0),
+    cuda::proclaim_return_type<char*>(
+      [inbuf = inbuf.begin(), inbuf_offsets = inbuf_offsets.begin()] __device__(size_t i) -> char* {
+        return &inbuf[inbuf_offsets[i]];
+      }));
+
+  // cub device batched copy
+  size_t temp_storage_bytes = 0;
+  cub::DeviceCopy::Batched(nullptr,
+                           temp_storage_bytes,
+                           input_it,
+                           output_it,
+                           col_lengths.begin(),
+                           col_lengths_size,
+                           stream.value());
+  rmm::device_buffer temp_storage(temp_storage_bytes, stream);
```

Comment:
we can clear temp_storage after the second cub::DeviceCopy::Batched call

Reply:
Moved the cub calls to a block.